### PR TITLE
attrib parsing and rendering

### PR DESCRIPTION
**Pull request by kimkanghyune**
aec original
aec dxf-viewer

As you can see in the image above, it is now fully rendered.
The previous pull request has been closed.

I used [this](https://drive.google.com/file/d/1keX3u87Idbop3-ENwdDk4iFZ8TUBWp4a/view?usp=sharing) example.

### CODE REVIEW

**Comment by kimkanghyune** [comment]
This ATTRIB with dimension lines does not load the layers correctly.
Anyone have any guesses?

**Comment by vagran** [comment]
Just tried to test it:
```
DxfWorker.js?7c56:66 TypeError: Cannot read properties of undefined (reading 'get')
    at DxfScene._DecomposeATTRIB (DxfScene.js?feb7:583:1)
    at _DecomposeATTRIB.next (<anonymous>)
    at DxfScene._ProcessDxfEntity (DxfScene.js?feb7:288:1)
    at DxfScene.Build (DxfScene.js?feb7:148:1)
    at async DxfWorker._Load (DxfWorker.js?7c56:159:1)
    at async DxfWorker._ProcessRequestMessage (DxfWorker.js?7c56:80:1)
    at async DxfWorker._ProcessRequest (DxfWorker.js?7c56:64:1)
```

File: src/DxfScene.js [hunk 1]
```diff
@@ -570,6 +573,29 @@ export class DxfScene {
             lineType: null
         })
     }
+    *_DecomposeATTRIB(entity, blockCtx) {
```
Comment (vagran) [inline src/DxfScene.js:583]: Please follow existing naming convention, no capitalized text: `_DecomposeAttrib`

**Comment by kimkanghyune** [comment]
It was a problem caused by fontStyle not being defined, and it was quickly fixed.

File: src/parser/DxfParser.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ import AUTO_CAD_COLOR_INDEX from './AutoCadColorIndex';
 import Face from './entities/3dface';
 import Arc from './entities/arc';
 import AttDef from './entities/attdef';
+import AttRib from './entities/attrib'
```
Comment (vagran) [inline src/parser/DxfParser.js:7]: Please, rename to `Attrib`. `AttDef` is short for "Attribute Definition", `Attrib` is for "Attribute"

**Comment by kimkanghyune** [comment]
> Just tried to test it:
> 
> ```
> DxfWorker.js?7c56:66 TypeError: Cannot read properties of undefined (reading 'get')
>     at DxfScene._DecomposeATTRIB (DxfScene.js?feb7:583:1)
>     at _DecomposeATTRIB.next (<anonymous>)
>     at DxfScene._ProcessDxfEntity (DxfScene.js?feb7:288:1)
>     at DxfScene.Build (DxfScene.js?feb7:148:1)
>     at async DxfWorker._Load (DxfWorker.js?7c56:159:1)
>     at async DxfWorker._ProcessRequestMessage (DxfWorker.js?7c56:80:1)
>     at async DxfWorker._ProcessRequest (DxfWorker.js?7c56:64:1)
> ```

It was a problem caused by fontStyle not being defined, and it was quickly fixed.

File: src/DxfScene.js [hunk 3]
```diff
@@ -570,6 +573,29 @@ export class DxfScene {
...
+        if (!this.textRenderer.canRender) {
+            return;
+        }
+        const layer = this._GetEntityLayer(entity, blockCtx);
+        const color = this._GetEntityColor(entity, blockCtx);
+
+        const font = this.fontStyles.get(entity.textStyle);
```
Comment (vagran) [inline src/DxfScene.js:590]: This causes exception now. `fontStyles` are not longer needed?

**Review by vagran — CHANGES_REQUESTED** [review]

**Comment by vagran** [comment]
> > Just tried to test it:
> > ```
> > DxfWorker.js?7c56:66 TypeError: Cannot read properties of undefined (reading 'get')
> >     at DxfScene._DecomposeATTRIB (DxfScene.js?feb7:583:1)
> >     at _DecomposeATTRIB.next (<anonymous>)
> >     at DxfScene._ProcessDxfEntity (DxfScene.js?feb7:288:1)
> >     at DxfScene.Build (DxfScene.js?feb7:148:1)
> >     at async DxfWorker._Load (DxfWorker.js?7c56:159:1)
> >     at async DxfWorker._ProcessRequestMessage (DxfWorker.js?7c56:80:1)
> >     at async DxfWorker._ProcessRequest (DxfWorker.js?7c56:64:1)
> > ```
> 
> It was a problem caused by fontStyle not being defined, and it was quickly fixed.

Still failing:
```
TypeError: Cannot read properties of undefined (reading 'widthFactor')
    at DxfScene._DecomposeATTRIB (DxfScene.js?feb7:600:1)
    at _DecomposeATTRIB.next (<anonymous>)
    at DxfScene._ProcessDxfEntity (DxfScene.js?feb7:295:1)
    at DxfScene.Build (DxfScene.js?feb7:155:1)
    at async DxfWorker._Load (DxfWorker.js?7c56:159:1)
    at async DxfWorker._ProcessRequestMessage (DxfWorker.js?7c56:80:1)
    at async DxfWorker._ProcessRequest (DxfWorker.js?7c56:64:1)
```
Seems you do not handle a situation, when the font style is not found. BTW shouldn't it be found in that file? May be it is worth to double check why it happens?

**Comment by vagran** [comment]
It still missing some critical part - preprocessing text entities in _FetchFonts() but I will add it myself.

**Pull request merged**